### PR TITLE
constant propogation will work for static constants in php

### PR DESCRIPTION
**Pull request by wu-s-john**
- With changes in the php ast in the pfff code, self is now a special identifier for expressions. As a result, we can use the self's properties to ensure that we can retrieve global constants within a class. This can help identify constants for constant propogation in php

- Pfff changes are made here https://github.com/returntocorp/pfff/pull/516
PR checklist:

- [X] Documentation is up-to-date
- [ ] Changelog is up-to-date
- [X] Change has no security implications (otherwise, ping security team)

### CODE REVIEW

File: semgrep-core/src/analyzing/Constant_propagation.ml [hunk 1]
```diff
@@ -208,6 +208,10 @@ let rec eval env x : svalue option =
   | N (Id (_, { id_svalue = { contents = Some x }; _ }))
   | DotAccess
       ( { e = IdSpecial (This, _); _ },
```
Comment (IagoAbal) [inline semgrep-core/src/analyzing/Constant_propagation.ml]: It's better to just write `This | Self` here, so:
```ocaml
| DotAccess
       ( { e = IdSpecial (This | Self, _); _ },
```

File: semgrep-core/src/analyzing/Constant_propagation.ml [hunk 2]
```diff
@@ -568,6 +572,7 @@ let propagate_basic lang prog =
               let/ svalue = find_id env id id_info in
               id_info.id_svalue := Some svalue
           | DotAccess ({ e = IdSpecial (This, _); _ }, _, FN (Id (id, id_info)))
```
Comment (IagoAbal) [inline semgrep-core/src/analyzing/Constant_propagation.ml]: Same here.

File: semgrep-core/src/naming/Naming_AST.ml [hunk 3]
```diff
@@ -746,6 +748,7 @@ let resolve lang prog =
                     error tok (spf "could not find '%s' in environment" s));
               recurse := false
           | DotAccess ({ e = IdSpecial (This, _); _ }, _, FN (Id (id, id_info)))
+          | DotAccess ({ e = IdSpecial (Self, _); _ }, _, FN (Id (id, id_info)))
```
Comment (IagoAbal) [inline semgrep-core/src/naming/Naming_AST.ml]: Same here.

**Comment by IagoAbal** [comment]
Thanks for your time fixing this issue @wu-s-john ! 🚀 I will be adding some comments in both your PRs.

**Comment by IagoAbal** [comment]
Looks good to me!

**Comment by wu-s-john** [comment]
@IagoAbal nice trick in OCaml. I love it. Addressed your comments. Please review again

**Comment by aryx** [comment]
This needs to be remerged with the latest develop to avoid the conflicts.

**Comment by wu-s-john** [comment]
@aryx @IagoAbal 

The branch has been updated and the checks have been completed. Can you please review again?

**Comment by aryx** [comment]
@spencerdrak can you investigate why the pre-commit jobs seems blocked? Is it because john is an external contributor?

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
Looks great! Thx a lot.
Feel free to update Changelog.log to explain your fix (and take credit for it).

**Pull request merged**